src/clj/puppetlabs/services/jruby/jruby_core.clj

```diff
@@ -204,11 +204,10 @@
   timeout. If the timeout runs out then nil will be returned, indicating that
   there were no instances available."
   [pool-context :- jruby-schemas/PoolContext
-   timeout :- schema/Int
```

Comment:
>Was there anything else that motivated this change, or was it just an inconsistency that you noticed?

Yeah, I just took it out because it seemed like an inconsistency and like it would be easier for users not to have to pass it in.

It's not used elsewhere - in puppet server we take it out of the config, stuff it into the context, and then take it out of the context to pass to this function in the service's `borrow-instance`.

IMO it seems like part of the config - it gets read out of the config file or the default is inserted via `initialize-config`, so removing it from the config after the fact seems weird to me.

I definitely agree there's a use case for it, even if we don't have that use case yet in puppetserver. I'm fine with reverting this change, adding another arity, or adding a `borrow-from-pool-with-custom-timeout` function. I think I would prefer either of the last two.